### PR TITLE
Fix validation issue of Get url maximum length

### PR DESCRIPTION
**Pull request by HyeongBaeYu**
When agents are generated too many, Get url will be longer. But Get query maximum length should be 2048.
If length is over 2048, google responses error and  Agent Event Flow shows nothing. so I add error message at this case.

### CODE REVIEW

File: app/helpers/dot_helper.rb [hunk 1]
```diff
@@ -8,9 +8,15 @@ def render_agents_diagram(agents)
         } rescue false)
       decorate_svg(svg, agents).html_safe
     else
-      tag('img', src: URI('https://chart.googleapis.com/chart').tap { |uri|
-            uri.query = URI.encode_www_form(cht: 'gv', chl: agents_dot(agents))
-          })
+	uriquery=URI.encode_www_form(cht: 'gv', chl: agents_dot(agents))
+	#Get query maximum length should be under 2048 bytes with including "chart?" of google chart request url
+	if uriquery.length > 2042
```
Comment (cantino) [inline app/helpers/dot_helper.rb]: Thanks @HyeongBaeYu!

Can you please fix the indentation so that it's always 2 spaces?

Comment (HyeongBaeYu) [inline app/helpers/dot_helper.rb]: Thanks, cantino!
I updated this fix. Please review it.

File: app/helpers/dot_helper.rb [hunk 2]
```diff
@@ -8,9 +8,15 @@ def render_agents_diagram(agents)
         } rescue false)
       decorate_svg(svg, agents).html_safe
     else
-      tag('img', src: URI('https://chart.googleapis.com/chart').tap { |uri|
-            uri.query = URI.encode_www_form(cht: 'gv', chl: agents_dot(agents))
-          })
+      uriquery=URI.encode_www_form(cht: 'gv', chl: agents_dot(agents))
+      #Get query maximum length should be under 2048 bytes with including "chart?" of google chart request url
+      if uriquery.length > 2042
+        "Too many agent to display, please check unused agents"
```
Comment (faizalzakaria) [inline app/helpers/dot_helper.rb]: I don't know if we already support this, but would it be great if we use [internationalization](http://guides.rubyonrails.org/i18n.html) ?

File: app/helpers/dot_helper.rb [hunk 3]
```diff
@@ -8,9 +8,15 @@ def render_agents_diagram(agents)
         } rescue false)
       decorate_svg(svg, agents).html_safe
     else
-      tag('img', src: URI('https://chart.googleapis.com/chart').tap { |uri|
-            uri.query = URI.encode_www_form(cht: 'gv', chl: agents_dot(agents))
-          })
+      uriquery=URI.encode_www_form(cht: 'gv', chl: agents_dot(agents))
```
Comment (faizalzakaria) [inline app/helpers/dot_helper.rb]: looks like bash code style. would it be cleaner as such, `uriquery = URI.encode_www_form(cht: 'gv', chl: agents_dot(agents))` ?

File: app/helpers/dot_helper.rb (hunk 2 above)
Comment (cantino) [inline app/helpers/dot_helper.rb]: Hey @faizalzakaria, we don't have that yet, and I don't think this PR is the right place to add it.  I would support adding it as a new effort, however!

Comment (faizalzakaria) [inline app/helpers/dot_helper.rb]: Got it @cantino , thanks.

File: app/helpers/dot_helper.rb (hunk 3 above)
Comment (HyeongBaeYu) [inline app/helpers/dot_helper.rb]: Hello @faizalzakaria,
I am beginner of rails. Could you explain the more detailed?

Comment (cantino) [inline app/helpers/dot_helper.rb]: I think @faizalzakaria means that our typical style would have spaces around the equal sign, so

```ruby
uri_query = URI...
```

File: app/helpers/dot_helper.rb (hunk 2 above)
Comment (cantino) [inline app/helpers/dot_helper.rb]: Maybe "There are too many Agents to display.  Can you remove some unused Agents?"

File: app/helpers/dot_helper.rb (hunk 3 above)
Comment (HyeongBaeYu) [inline app/helpers/dot_helper.rb]: Thanks @cantino for advice.
I added spaces around the equal sign for code style.

**Comment by HyeongBaeYu** [comment]
After I added just space for code style, the Travis CI build was failed. Do I need more another action? Please advise me.

**Comment by cantino** [comment]
Thanks @HyeongBaeYu!  No, that was a random failure.  Looks great and has been merged, much appreciated!

**Pull request merged**